### DISCUSSION TITLE
Power on/off for specific time

### DISCUSSION THREAD

**Discussion by Rudi-Ment** [arendst/Tasmota#19728] [Q&A] [answered]
Hi.
I have the following wish:
I need a power on for a certain time.
Maybe a command like power<x> on 300 for a power on for 300 secounds or similar.
The same for off an toggle for compled the function.

So you have a pulsetime from 7200 as a safety off a can switch on from a source like iob or something for 300 secounds. From an other channel or button you can switch on for 600 secounds and so on.
Than the switch off is on the hardware. In case of a connection error, a switch of is safe.
The Homematic components have this function two and so you can have different times from any switch-command-source.

Greating

Rüdiger Haack

**Comment by sfromis** [comment]
If you already use PulseTime, additional timing can be done using rules with a RuleTimer.

**Reply by Rudi-Ment** [reply]
Can you please write down an example here ?
I want to use three different times.
1. a garagedoor opener via script switch on the light for 5 minutes
2. a buttonswitch switch on the light for 1 hour via mqqt or http-request.
3. a motionsensor switch it on for 1 minute via script or athor method.
4. a savety time is the pulsetime with about 1 hour

**Reply by sfromis** [reply]
See the docs for how it [RuleTimer](https://tasmota.github.io/docs/Commands/#ruletimer) works.
https://tasmota.github.io/docs/Rules/#time-delayed-auto-off-switch

**Reply by Rudi-Ment** [reply]
I know the rules like that. i mean this cases.
But how can i use the different times ?
How can i use three different triggers for the rules, like my example above ?
In the rule is not thinking over different times in different triggers.

I just need a chance to give the tasmota switch a time and than a switch command (where the time is only for the following sitchcommand) or a combination of switch and time (like "power1 1 130" for 30 second on).

**Reply by sfromis** [reply, marked as answer]
The basic answer is still to create rules, like using the `Event` command and rule triggers to match, with the time as a parameter. One event/rule per power channel, if you want multiple channels.
Something like `Event p2=300` for 5 minutes (the value you use in the RuleTimer command from the rule picking up the Event). Those Event commands can come via MQTT, console, or other rules.

**Reply by sfromis** [reply]
I suppose that you're not using an ESP32 device, but just in case....
There you can easily use Berry to create additional commands, like this TimedPower command:
```
TimedPower3 10
TimedPower4 20,blink
``` 

```
# https://tasmota.github.io/docs/Berry/#functions-to-create-custom-tasmota-command
tasmota.remove_cmd('TimedPower')
tasmota.add_cmd('TimedPower', 
  def (cmd, idx, payload)
    import string, json
    var sp = string.split(payload, ',')
    var after = sp.size()>0 ? (size(sp[0])>0 ? real(sp[0]) : 10) : 10
    var parm = sp.size()>1 ? sp[1] : 'on'
    tasmota.cmd(f'Power{idx} {parm}')
    var offcmd = f'Power{idx} off'
    var offtim = tasmota.time_str(tasmota.rtc()['local'] + int(after))
    tasmota.set_timer(int(after*1000), /->tasmota.cmd(offcmd), f'TimedPower {idx}')
    tasmota.resp_cmnd(json.dump({cmd:f'Will {offcmd} at {offtim}'}))
  end)
```

**Reply by Rudi-Ment** [reply]
The hint with the events is what i have missed.
I wanna try.
Thank you

**Comment by markusb68** [comment]
Hello saw this is added to v13.4.0 Quinta:
`Command TimedPower<index> <milliseconds>[,ON|OFF|TOGGLE|BLINK] executes Power<index> [ON|OFF|TOGGLE|BLINK] and after executes Power<index> [OFF|ON|TOGGLE|OFF]`

Please describe exactly what TimedPower os doing. Can't find it in the Commands: [https://tasmota.github.io/docs/Commands/](url)

I'm especially looking for something similar to pulsetime, but not exactly.
I want a command to keep an relay AT LEAST on for n seconds. So when this command is set a POWER OFF should NOT turn the relay OFF during the first n Seconds.

Any Idea how to solve it on an Simpe SONOFF (ESP8266, NOT ESP32, no BARRY) ?

**Reply by sfromis** [reply]
It is like the normal `Power` command, just with addition of a time in milliseconds as the first argument. After that interval, the action will be reverted. See:
https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_support/support_command.ino#L727-L740
(And it is available across chip types, does not depend on Berry)

**Reply by sfromis** [reply]
Tasmota has no feature to disregard command you give it, and what works best depends on the exact details of how all combinations of controls should work. You can manage a lot with rules, especially if you vary a bit in how to control things.

**Comment by lalo-uy** [comment]
You can do that seting an unused gpio for the comands, an catching the
on/off in rules. There you set the timer and real gpio.
